plugin/storage/es/factory.go

```diff
@@ -96,45 +101,48 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 	if err != nil {
 		return fmt.Errorf("failed to create primary Elasticsearch client: %w", err)
 	}
-	f.primaryClient = primaryClient
+	f.atomicPrimaryClient.Store(&primaryClient)
+
 	if f.archiveConfig.Enabled {
-		f.archiveClient, err = f.newClientFn(f.archiveConfig, logger, metricsFactory)
+		archiveClient, err := f.newClientFn(f.archiveConfig, logger, metricsFactory)
 		if err != nil {
 			return fmt.Errorf("failed to create archive Elasticsearch client: %w", err)
 		}
+		f.atomicArchiveClient.Store(&archiveClient)
 	}
+
 	return nil
 }
 
 // CreateSpanReader implements storage.Factory
 func (f *Factory) CreateSpanReader() (spanstore.Reader, error) {
-	return createSpanReader(f.metricsFactory, f.logger, f.primaryClient, f.primaryConfig, false)
+	return createSpanReader(f.metricsFactory, f.logger, *f.atomicPrimaryClient.Load(), f.primaryConfig, false)
```

Comment:
You cannot dereference client here, if you do it will not be swappable once the reader is created. You need to pass the whole atomic var.

This indicates that you don't have a unit test which validates this new capability of updating password via file.